### PR TITLE
Add support for proxy and airgap installs

### PR DESCRIPTION
**Pull request by gene-redpanda**
This PR makes numerous changes to support proxy installs and airgap. Chief among them are

* adding support for proxy based install and proxy configuration
* adding system_setup role
* adding binary_bundler role
* adding the taskfiles necessary to support installing with binaries

Adding support for proxy based install and proxy configuration was a necessary step for achieving our end goal of an airgapped install as it allowed us to created a networking constrained environment where soup to nuts testing could occur. It also allows us to properly support a standard model for production deployments. Squid proxy was chosen as it is a fairly standard, well understood linux proxy that supports http, https and ftp proxying. While ftp proxying is not necessary for redpanda installs directly it is essential to support apt/dnf updates as a nontrivial number of standard repositories used in these updates are accessed via ftp. 

Breaking out system setup allows us to better control the order of operations that need to occur in a playbook for a redpanda cluster to be created in either a proxy or airgap configuration. It ensures that for proxy config we've properly handled the apt/dnf proxy before attempting to manage certs and installs, and for an airgap install it allows us to simply drop this step as unncessary.

binary bundler by contrast provides a quick and easy process for generating a bundle of the necessary RPMs/DEBs for a redpanda install given the necessary inputs (which are checked with asserts).

Finally the airgap install taskfiles in the broker role allow us to ensure that we are properly handling the copy and install of the DEB/RPM bundles, including the all important chowning to redpanda of the data directory for the dpkg install in particular.

### CODE REVIEW

File: roles/redpanda_broker/tasks/install-rp-deb.yml
Comment (r-vasquez) [inline roles/redpanda_broker/tasks/install-rp-deb.yml:36]: [Question] why is the `\n` needed?

File: roles/binary_bundler/tasks/main.yml
Comment (gene-redpanda) [reply roles/binary_bundler/tasks/main.yml:2]: Yes actually.

File: roles/system_setup/README.md
Comment (twmb) [reply roles/system_setup/README.md:4]:
```
This entails:
*
*
*
```

File: roles/redpanda_broker/tasks/main.yml
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/main.yml:5]: Yes, versions can still be specified. I just removed the default because I wanted to make sure users actually picked the version rather than using our default. I'll toss in another assert to ensure it is specified.

File: roles/system_setup/README.md
Comment (r-vasquez) [reply roles/system_setup/README.md:4]: Oh I was referring also to `popping certs in` maybe to `installing certificates` or something along those lines. :sweat_smile:

Comment (gene-redpanda) [reply roles/system_setup/README.md]: Ok, changed that too

**Review by r-vasquez — APPROVED** [review]

**Comment by hcoyote** [comment]
I'm asking we hold this PR until we've had further discussion on it. I'm not sure about the direction being taken here with this binary bundler. @deniscoady @gene-redpanda @tmgstevens

**Comment by gene-redpanda** [comment]
> I'm asking we hold this PR until we've had further discussion on it. I'm not sure about the direction being taken here with this binary bundler. @deniscoady @gene-redpanda @tmgstevens

Can you provide some more info on your issues with it? It pretty much exactly does what we'd discussed: downloads the binaries and dumps them in a tarball for copying into the brokers.

**Comment by gene-redpanda** [comment]
> I'm asking we hold this PR until we've had further discussion on it. I'm not sure about the direction being taken here with this binary bundler. @deniscoady @gene-redpanda @tmgstevens

@hcoyote mind taking a peek now that we've had our discussion? I've got the discussed doc in process 😄

File: roles/redpanda_broker/defaults/main.yml
Comment (hcoyote) [inline roles/redpanda_broker/defaults/main.yml:65]: these two are still needed for when the user wants to use our separate playbook for setting up /var/lib/redpanda on linux raid, right? Did this get moved into that playbook directly?

Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml:65]: Bah, the one thing I didn't test for RAID. I'll readd it

File: roles/redpanda_broker/tasks/install-rp-deb-airgap.yml
Comment (hcoyote) [inline roles/redpanda_broker/tasks/install-rp-deb-airgap.yml:15]: we should enforce a permission here. at very least, 0770.

Comment (hcoyote) [reply roles/redpanda_broker/tasks/install-rp-deb-airgap.yml:15]: I think /var/lib/redpanda should be empty at this point, not sure we need to recurse to set everything. this could cause long chown runs once the directory is fully populated.

File: roles/redpanda_broker/tasks/main.yml
Comment (hcoyote) [inline roles/redpanda_broker/tasks/main.yml:6]: we've lost context in the error message I think. What should an example redpanda version number be (or latest)?

File: roles/redpanda_broker/defaults/main.yml
Comment (hcoyote) [reply roles/redpanda_broker/defaults/main.yml:65]: I guess all this stuff got moved to system_setup?

File: roles/system_setup/tasks/install-node-deps-rpm.yml
Comment (hcoyote) [inline roles/system_setup/tasks/install-node-deps-rpm.yml:21]: the more I think about this, the more I wonder if we should be nuking repos on behalf of the OS admin. I'm not sure the origin of these.

Comment (hcoyote) [reply roles/system_setup/tasks/install-node-deps-rpm.yml:21]: this is not a blocker, we can handle removal in a separate PR

**Review by hcoyote — APPROVED** [review]
LGTM pending the permission/ recurssive owner setting

File: roles/redpanda_broker/defaults/main.yml
Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml:65]: Ah yeah sorry I misread the file name, they're still there but in system setup. The goal is to ensure end users can have an easier process for deciding what, exactly, we do with their nodes while still providing a soup to nuts option for end users that aren't as concerned.

File: roles/redpanda_broker/tasks/install-rp-deb-airgap.yml
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-rp-deb-airgap.yml:15]: Ansible defaults to enforcing the system default permissions which IMO is preferable here since we don't have a specific need.

File: roles/system_setup/tasks/install-node-deps-rpm.yml
Comment (gene-redpanda) [reply roles/system_setup/tasks/install-node-deps-rpm.yml:21]: My thought is we're doing it and we probably have a reason why we're doing it 🤷‍♀️ so until someone complains I'm leaving it in.

**Pull request merged**
